### ISSUE TITLE
Configurable CodyChat window width

### ISSUE THREAD

**Issue by audkar** [sourcegraph/sg.nvim#178]
#### Problem:

Chat window is too small. After increasing the window size, closing the CodyChat window (`:CodyToggle`) and reopening it, the size is reset. This is very inconvenient. 

#### Feature request:

Allow to configure default CodyChat window width. Or remember last used width. Or both 

:bow:

**Comment by Odas0R** [comment]
> #### Problem:
> Chat window is too small. After increasing the window size, closing the CodyChat window (`:CodyToggle`) and reopening it, the size is reset. This is very inconvenient.
> 
> #### Feature request:
> Allow to configure default CodyChat window width. Or remember last used width. Or both
> 
> 🙇

+ 1
The plugin should at least store the previous window width, by default

Great work @tjdevries, big fan :muscle:

**Comment by smackesey** [comment]
Yes please-- the window really should be more configurable all around (esp to be floating). I use Neovim on a 34-inch display and the tiny non-floating window on the side that jolts all my other open windows makes the chat quite unpleasant to use.

[ChatGPT.nvim](https://github.com/jackMort/ChatGPT.nvim) can provide some inspiration here.

**Comment by tjdevries** [comment]
```lua
-- this API might not be the best, but just to share the idea.
require("sg").setup({
 layouts = {
   chat = { ... }
   task = { ... }
 }
})
```

Yes, i think something along these lines would be good (and possibly more advanced floating style ones like chatgpt.nvim, which looks quite nice). I'm currently working on a larger update to the backend of the chat experience, so once we clean that up it should be alot easier to do different stuff for the various layouts (possibly providing custom layouts as well)
